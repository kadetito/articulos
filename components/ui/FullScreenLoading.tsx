import React from "react";

export const FullScreenLoading = () => {
  return (
    <div
      aria-label="loader"
      id="pause"
      className="d-flex align-items-center justify-content-center"
    >
      <div id="spinner"></div>
    </div>
  );
};
