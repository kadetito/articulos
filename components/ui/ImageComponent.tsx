import { Card } from "react-bootstrap";
import { FC } from "react";
import { LazyLoadImage } from "react-lazy-load-image-component";
import "react-lazy-load-image-component/src/effects/blur.css";

interface Props {
  title: string;
  image: string;
  footImages: string;
}

export const ImageComponent: FC<Props> = ({ title, image, footImages }) => {
  return (
    <>
      <LazyLoadImage
        src={image}
        alt={title}
        className="img-thumbnail mb-1"
        effect="blur"
      />
      <p>
        <em>{footImages}</em>
      </p>
    </>
  );
};
