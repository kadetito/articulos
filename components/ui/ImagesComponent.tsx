import { FC } from "react";
import { LazyLoadImage } from "react-lazy-load-image-component";
import "react-lazy-load-image-component/src/effects/blur.css";

interface Props {
  title: string;
  images: string[];
  footImages: string[];
}

export const ImagesComponent: FC<Props> = ({ title, images, footImages }) => {
  console.log(images);

  return (
    <>
      {images.map((image: any, index: any) => (
        <div className="mb-2" key={index}>
          <LazyLoadImage
            src={`../articles/${image}`}
            alt={title}
            className="img-thumbnail mb-1"
            effect="blur"
          />
          <p>
            <em>{footImages}</em>
          </p>
        </div>
      ))}
    </>
  );
};
