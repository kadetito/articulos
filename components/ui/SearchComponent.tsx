import React from "react";
import { Form, Button } from "react-bootstrap";
import { useState } from "react";
import { useRouter } from "next/router";

export const SearchComponent = () => {
  const router = useRouter();
  const [searchterm, setSearchterm] = useState("");

  const onSearchTerm = () => {
    if (searchterm.trim().length === 0) return;
    navigateTo(`/search/${searchterm}`);
  };

  const navigateTo = (url: string) => {
    router.push(url);
  };

  return (
    <>
      <input
        type="search"
        data-testid="inputsearch"
        value={searchterm}
        onChange={(e) => setSearchterm(e.target.value)}
        onKeyPress={(e) => e.key === "Enter" && onSearchTerm()}
        placeholder="Search"
        className="me-2 form-control"
        aria-label="Search"
      />
      <Button
        data-testid="buttonsearch"
        onClick={onSearchTerm}
        variant="outline-secondary"
        size="sm"
        className="ms-2"
      >
        Search
      </Button>
    </>
  );
};
