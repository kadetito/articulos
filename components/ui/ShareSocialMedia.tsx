import { useRouter } from "next/router";
import {
  EmailShareButton,
  EmailIcon,
  FacebookShareButton,
  FacebookIcon,
  LinkedinShareButton,
  LinkedinIcon,
  TelegramShareButton,
  TelegramIcon,
  TwitterShareButton,
  TwitterIcon,
  WhatsappShareButton,
  WhatsappIcon,
} from "react-share";

export const ShareSocialMedia = () => {
  const { asPath } = useRouter();
  const shareUrl = `${process.env.NEXT_PUBLIC_API_URL}${asPath}`;

  return (
    <>
      <EmailShareButton
        className="me-2"
        children={<EmailIcon size={32} round={true} />}
        url={shareUrl}
      />
      <FacebookShareButton
        className="me-2"
        children={<FacebookIcon size={32} round={true} />}
        url={shareUrl}
      />

      <TwitterShareButton
        className="me-2"
        children={<TwitterIcon size={32} round={true} />}
        url={shareUrl}
      />
      <LinkedinShareButton
        className="me-2"
        children={<LinkedinIcon size={32} round={true} />}
        url={shareUrl}
      />
      <TelegramShareButton
        className="me-2"
        children={<TelegramIcon size={32} round={true} />}
        url={shareUrl}
      />
      <WhatsappShareButton
        className="me-2"
        children={<WhatsappIcon size={32} round={true} />}
        url={shareUrl}
      />
    </>
  );
};
