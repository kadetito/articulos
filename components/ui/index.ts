export * from "./NavBar";
export * from "./ImageComponent";
export * from "./ImagesComponent";
export * from "./ShareSocialMedia";
export * from "./FullScreenLoading";
export * from "./SearchComponent";
