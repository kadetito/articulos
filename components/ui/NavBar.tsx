import NextLink from "next/link";
import { useRouter } from "next/router";
import { Form } from "react-bootstrap";
import Button from "react-bootstrap/Button";
import Container from "react-bootstrap/Container";
import Nav from "react-bootstrap/Nav";
import Navbar from "react-bootstrap/Navbar";
import Offcanvas from "react-bootstrap/Offcanvas";
import { SearchComponent } from "./SearchComponent";

export const NavBar = () => {
  const { asPath } = useRouter();
  return (
    <>
      <Navbar key="sm" bg="dark" expand="sm" className="mb-3">
        <Container fluid>
          <Navbar.Brand>
            <NextLink href="/" passHref>
              Navbar Offcanvas
            </NextLink>
          </Navbar.Brand>
          <Navbar.Toggle aria-controls={`offcanvasNavbar-expand-$"sm"`} />
          <Navbar.Offcanvas
            id={`offcanvasNavbar-expand-$"sm"`}
            aria-labelledby={`offcanvasNavbarLabel-expand-$"sm"`}
            placement="end"
          >
            <Offcanvas.Header closeButton>
              <Offcanvas.Title id={`offcanvasNavbarLabel-expand-$"sm"`}>
                Offcanvas
              </Offcanvas.Title>
            </Offcanvas.Header>
            <Offcanvas.Body>
              <Nav className="justify-content-end flex-grow-1 pe-3">
                <Nav>
                  <NextLink href="/category/news" passHref>
                    <Button
                      variant={
                        asPath === "/category/news"
                          ? "secondary"
                          : "outline-secondary"
                      }
                      size="sm"
                      className="ms-2"
                    >
                      Novedades
                    </Button>
                  </NextLink>
                </Nav>
                <Nav>
                  <NextLink href="/category/opinion" passHref>
                    <Button
                      variant={
                        asPath === "/category/opinion"
                          ? "secondary"
                          : "outline-secondary"
                      }
                      size="sm"
                      className="ms-2"
                    >
                      Opini√≥n
                    </Button>
                  </NextLink>
                </Nav>
                <Nav>
                  <NextLink href="/category/advices" passHref>
                    <Button
                      variant={
                        asPath === "/category/advices"
                          ? "secondary"
                          : "outline-secondary"
                      }
                      size="sm"
                      className="ms-2"
                    >
                      Consejos
                    </Button>
                  </NextLink>
                </Nav>

                {/* <NavDropdown
                  title="Dropdown"
                  id={`offcanvasNavbarDropdown-expand-$"sm"`}
                >
                  <NavDropdown.Item href="#action3">Action</NavDropdown.Item>
                  <NavDropdown.Item href="#action4">
                    Another action
                  </NavDropdown.Item>
                  <NavDropdown.Divider />
                  <NavDropdown.Item href="#action5">
                    Something else here
                  </NavDropdown.Item>
                </NavDropdown> */}
              </Nav>
              <div className="d-flex">
                <SearchComponent />
              </div>
            </Offcanvas.Body>
          </Navbar.Offcanvas>
        </Container>
      </Navbar>
    </>
  );
};
