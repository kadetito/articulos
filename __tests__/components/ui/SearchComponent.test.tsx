import { screen, render, fireEvent } from "@testing-library/react";

import { SearchComponent } from "../../../components/ui/SearchComponent";

jest.mock("next/router", () => ({
  useRouter: jest.fn(),
}));

describe("Search Component", () => {
  it("<SearchComponent /> debe hacer match con el snapshot", () => {
    const { container } = render(<SearchComponent />);
    expect(container).toMatchSnapshot();
  });

  test("Search input get value", () => {
    render(<SearchComponent />);
    const searchInput = screen.getByTestId("inputsearch") as HTMLInputElement;
    fireEvent.input(searchInput, { target: { value: "test" } });
    expect(searchInput.value).toEqual("test"); // OR
  });

  test("onSearchTerm no devuelve nada", () => {
    const signinMock = jest.fn();
    const props: any = {
      signin: signinMock,
    };
    const component = render(<SearchComponent {...props} />);
    component.getByTestId("buttonsearch").click();
    setTimeout(() => {
      expect(signinMock.mock.calls.length).to.equal("");
    }, 0);
  });
});
