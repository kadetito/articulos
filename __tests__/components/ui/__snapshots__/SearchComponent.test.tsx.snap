// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component <SearchComponent /> debe hacer match con el snapshot 1`] = `
<div>
  <input
    aria-label="Search"
    class="me-2 form-control"
    data-testid="inputsearch"
    placeholder="Search"
    type="search"
    value=""
  />
  <button
    class="ms-2 btn btn-outline-secondary btn-sm"
    data-testid="buttonsearch"
    type="button"
  >
    Search
  </button>
</div>
`;
